src/Examples/Advanced/CustomProperties/Hydrocarbon_Processing_example.ipynb

```diff
@@ -0,0 +1,1078 @@
+      "   inequality constraints with lower and upper bounds:        0\n",
+      "        inequality constraints with only upper bounds:        0\n",
+      "\n",
+      "iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls\n",
+      "   0  0.0000000e+00 5.88e+05 1.00e+00  -1.0 0.00e+00    -  0.00e+00 0.00e+00   0\n",
+      "   1  0.0000000e+00 4.02e+05 4.10e+03  -1.0 8.22e+04    -  7.92e-01 3.17e-01h  1\n",
+      "   2  0.0000000e+00 2.15e+05 3.10e+03  -1.0 4.69e+02    -  9.90e-01 4.65e-01h  1\n",
+      "   3  0.0000000e+00 2.16e+03 2.15e+03  -1.0 2.40e+04    -  7.37e-01 9.90e-01h  1\n",
+      "   4  0.0000000e+00 2.10e+01 6.92e+03  -1.0 1.62e+03    -  6.45e-01 9.90e-01h  1\n",
+      "   5  0.0000000e+00 2.17e-01 8.71e+03  -1.0 3.46e+02    -  9.90e-01 1.00e+00h  1\n",
+      "   6  0.0000000e+00 4.28e-03 4.45e+04  -1.0 4.81e+01    -  9.95e-01 1.00e+00h  1\n",
+      "   7  0.0000000e+00 1.17e-06 1.68e+01  -1.0 7.94e-01    -  1.00e+00 1.00e+00h  1\n",
+      "\n",
+      "Number of Iterations....: 7\n",
+      "\n",
+      "                                   (scaled)                 (unscaled)\n",
+      "Objective...............:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Dual infeasibility......:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Constraint violation....:   4.0239024785785836e-09    1.1678785085678101e-06\n",
+      "Complementarity.........:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Overall NLP error.......:   4.0239024785785836e-09    1.1678785085678101e-06\n",
+      "\n",
+      "\n",
+      "Number of objective function evaluations             = 8\n",
+      "Number of objective gradient evaluations             = 8\n",
+      "Number of equality constraint evaluations            = 8\n",
+      "Number of inequality constraint evaluations          = 0\n",
+      "Number of equality constraint Jacobian evaluations   = 8\n",
+      "Number of inequality constraint Jacobian evaluations = 0\n",
+      "Number of Lagrangian Hessian evaluations             = 7\n",
+      "Total CPU secs in IPOPT (w/o function evaluations)   =      7.125\n",
+      "Total CPU secs in NLP function evaluations           =     39.292\n",
+      "\n",
+      "EXIT: Optimal Solution Found.\n"
+     ]
+    }
+   ],
+   "source": [
+    "result = solver.solve(m, tee=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "tags": [
+     "testing"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "from pyomo.opt import TerminationCondition, SolverStatus\n",
+    "\n",
+    "# Check if termination condition is optimal\n",
+    "assert result.solver.termination_condition == TerminationCondition.optimal\n",
+    "assert result.solver.status == SolverStatus.ok"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3.4 Results"
```

Comment:
A little more explanation of the results would probably be useful so the reader understands what is happening and why.